### PR TITLE
[2.7] Removed dead code and various cleaning

### PR DESCRIPTION
**Pull request by saro0h**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

In complement of these: https://github.com/symfony/symfony/pull/13488 & https://github.com/symfony/symfony/pull/13481

### CODE REVIEW

File: src/Symfony/Component/Form/Test/DeprecationErrorHandler.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
  */
 class DeprecationErrorHandler
 {
-    public static function handle($errorNumber, $message, $file, $line, $context)
+    public static function handle($errorNumber, $message, $file, $line)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Test/DeprecationErrorHandler.php]: Bc break, same below

File: src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php [hunk 2]
```diff
@@ -30,20 +30,11 @@ class DumpDataCollector extends DataCollector implements DataDumperInterface
...
     private $clonesIndex = 0;
-    private $rootRefs;
 
     public function __construct(Stopwatch $stopwatch = null, $fileLinkFormat = null)
     {
         $this->stopwatch = $stopwatch;
         $this->fileLinkFormat = $fileLinkFormat ?: ini_get('xdebug.file_link_format') ?: get_cfg_var('xdebug.file_link_format');
-
-        // All clones share these properties by reference:
-        $this->rootRefs = array(
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php]: Wrong, refs are required

Comment (saro0h) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php]: I don't understand :/ ``$rootRefs`` is private, there is no accessor and it's only used here. Where do you see the use of this?

Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php]: This is black magic with references: by holding references to other properties in the array, it turns these props to references. By doing so, all clones of a DumpDataCollector object share these props by reference. The comment was here to say that. Feel free to make it better :)

File: src/Symfony/Component/HttpKernel/Profiler/BaseMemcacheProfilerStorage.php [hunk 3]
```diff
@@ -154,7 +154,7 @@ public function write(Profile $profile)
         $data = array(
             'token' => $profile->getToken(),
             'parent' => $profile->getParentToken(),
-            'children' => array_map(function ($p) { return $p->getToken(); }, $profile->getChildren()),
+            'children' => array_map(function (Profile $p) { return $p->getToken(); }, $profile->getChildren()),
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Profiler/BaseMemcacheProfilerStorage.php]: This adds a type int check for no value imo

File: src/Symfony/Component/Debug/ExceptionHandler.php [hunk 4]
```diff
@@ -50,7 +50,7 @@ public function __construct($debug = true, $fileLinkFormat = null)
      */
     public static function register($debug = true, $fileLinkFormat = null)
     {
-        $handler = new static($debug, $fileLinkFormat = null);
+        $handler = new static($debug, $fileLinkFormat);
```
Comment (stof) [inline src/Symfony/Component/Debug/ExceptionHandler.php]: this is not a dead code removal. It is a bug fix. So it should go in older versions

File: src/Symfony/Component/Process/Process.php [hunk 5]
```diff
@@ -1296,7 +1296,7 @@ protected function buildCallback($callback)
         $that = $this;
         $out = self::OUT;
         $err = self::ERR;
```
Comment (stof) [inline src/Symfony/Component/Process/Process.php]: ``$err`` is not used anymore

File: src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php [hunk 6]
```diff
@@ -283,7 +283,7 @@ protected function instantiateObject(array $data, $class, array &$context, \Refl
                 } elseif ($constructorParameter->isOptional()) {
                     $params[] = $constructorParameter->getDefaultValue();
                 } else {
-                    throw new RuntimeException(
+                    throw new \RuntimeException(
```
Comment (stof) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: it should use the RuntimeException of the component instead (and this case should be covered by tests btw)

Comment (saro0h) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: I'll do another PR for the tests to not loose the "vacuum cleaner", if you don't mind :)

**Comment by fabpot** [comment]
Thank you @saro0h.

**Pull request merged**